pysat/utils/io.py

```diff
@@ -724,68 +1278,254 @@ def load_netcdf_xarray(fnames, strict_meta=False, file_format='NETCDF4',
     load_netcdf
 
     """
+
+    if decode_times is None:
+        dstr = ''.join(['Defaulting to `decode_times=False`. In the future ',
+                        'the default will be updated to `True`. Set a value ',
+                        'for `decode_times` to silence this warning.'])
+        warnings.warn(dstr, DeprecationWarning, stacklevel=2)
+        decode_times = False
+
+    if epoch_name is None:
+        epoch_name = 'time'
+
     # Ensure inputs are in the correct format
     fnames = pysat.utils.listify(fnames)
     file_format = file_format.upper()
 
     # Initialize local variables
     meta = pysat.Meta(labels=labels)
 
+    # Store all metadata in a dict that may be filtered before
+    # assignment to `meta`
+    full_mdict = {}
+
+    if meta_translation is None:
+        # Assign default translation using `meta`
+        meta_translation = default_from_netcdf_translation_table(meta)
+
+    # Drop metadata labels initialization
+    if drop_meta_labels is None:
+        drop_meta_labels = []
+    else:
+        drop_meta_labels = pysat.utils.listify(drop_meta_labels)
+
     # Load the data differently for single or multiple files
     if len(fnames) == 1:
-        data = xr.open_dataset(fnames[0], decode_timedelta=decode_timedelta)
+        data = xr.open_dataset(fnames[0], decode_timedelta=decode_timedelta,
+                               decode_times=decode_times)
     else:
         data = xr.open_mfdataset(fnames, decode_timedelta=decode_timedelta,
-                                 combine='by_coords')
-
-    # If epoch exists, convert to datetime object.
-    # TODO(#991): epoch does not necessarily exist.  May need to update if
-    # assumptions about time change.
-    if epoch_name in data.variables:
-        data[epoch_name] = xr.DataArray(pds.to_datetime(data[epoch_name],
-                                                        unit=epoch_unit,
-                                                        origin=epoch_origin),
-                                        coords=data[epoch_name].coords)
+                                 combine='by_coords', decode_times=decode_times)
+
+    # Need to get a list of all variables, dimensions, and coordinates.
+    all_vars = xarray_all_vars(data)
+
+    # Rename `epoch_name` to 'time'
+    if epoch_name != 'time':
+        if 'time' not in all_vars:
+            if epoch_name in data.dims:
+                data = data.rename({epoch_name: 'time'})
+            elif epoch_name in all_vars:
+                data = data.rename({epoch_name: 'time'})
+                wstr = ''.join(['Epoch label: "', epoch_name, '"',
+                                ' is not a dimension.'])
+                pysat.logger.warning(wstr)
+            else:
+                estr = ''.join(['Epoch label: "', epoch_name, '"',
```

Comment:
This was the one. Close enough to pass the same test. Will improve. Since the pandas ones already existed I'll use them as the reference.